lib/src/geometry/shape.dart

```diff
@@ -0,0 +1,88 @@
+import 'dart:ui';
+
+import '../../components.dart';
+import '../extensions/vector2.dart';
+import 'shape_intersections.dart' as intersection_system;
+
+abstract class Shape {
+  Vector2 position = Vector2.zero();
+  Vector2 size;
+  double angle = 0;
```

Comment:
I think some comments here would be good to explain how these concepts attach to arbitrary shapes

Reply:
Added, I hope they make sense.